packages/@uppy/provider-views/src/SearchInput.tsx

```diff
@@ -1,89 +1,57 @@
-/* eslint-disable react/require-default-props */
-import { h, Fragment } from 'preact'
-import { useEffect, useState, useCallback } from 'preact/hooks'
-import { nanoid } from 'nanoid/non-secure'
+import { h } from 'preact'
+import type { ChangeEvent } from 'preact/compat'
 
 type Props = {
-  search: (query: string) => void
-  searchOnInput?: boolean
-  searchTerm?: string | null
-  showButton?: boolean
+  searchString: string
+  setSearchString: (s: string) => void
+  submitSearchString: () => void
+
+  wrapperClassName: string
+  inputClassName: string
+
   inputLabel: string
   clearSearchLabel?: string
+
+  showButton?: boolean
   buttonLabel?: string
-  // eslint-disable-next-line react/require-default-props
-  clearSearch?: () => void
-  inputClassName: string
-  buttonCSSClassName?: string
 }
 
-export default function SearchFilterInput(props: Props) {
-  const {
-    search,
-    searchOnInput,
-    searchTerm,
-    showButton,
-    inputLabel,
-    clearSearchLabel,
-    buttonLabel,
-    clearSearch,
-    inputClassName,
-    buttonCSSClassName,
-  } = props
-  const [searchText, setSearchText] = useState(searchTerm ?? '')
-  // const debouncedSearch = debounce((q) => search(q), 1000)
+function SearchInput({
+  searchString,
+  setSearchString,
+  submitSearchString,
 
-  const validateAndSearch = useCallback(
-    (ev: Event) => {
-      ev.preventDefault()
-      search(searchText)
-    },
-    [search, searchText],
-  )
+  wrapperClassName,
+  inputClassName,
 
-  const handleInput = useCallback(
-    (ev: Event) => {
-      const inputValue = (ev.target as HTMLInputElement).value
-      setSearchText(inputValue)
-      if (searchOnInput) search(inputValue)
-    },
-    [setSearchText, searchOnInput, search],
-  )
+  inputLabel,
+  clearSearchLabel = '',
 
-  const handleReset = () => {
-    setSearchText('')
-    if (clearSearch) clearSearch()
+  showButton = false,
+  buttonLabel = '',
+}: Props) {
+  const onSubmit = (e: Event) => {
+    e.preventDefault()
+    submitSearchString()
   }
 
-  const [form] = useState(() => {
-    const formEl = document.createElement('form')
-    formEl.setAttribute('tabindex', '-1')
-    formEl.id = nanoid()
-    return formEl
-  })
-
-  useEffect(() => {
-    document.body.appendChild(form)
-    form.addEventListener('submit', validateAndSearch)
-    return () => {
-      form.removeEventListener('submit', validateAndSearch)
-      document.body.removeChild(form)
-    }
-  }, [form, validateAndSearch])
+  const onInput = (e: ChangeEvent) => {
+    setSearchString((e.target as HTMLInputElement).value)
+  }
 
   return (
-    <Fragment>
+    <form className={wrapperClassName} onSubmit={onSubmit}>
```

Comment:
We cannot do that, `Uppy` may be rendered inside another `<form>` element. (refs: https://github.com/transloadit/uppy/pull/4283). That's why we need to wtick with using `form` attribute

Reply:
Huh we do have forms in `AuthView.tsx` still

https://github.com/transloadit/uppy/blob/5695299a195f10c2e5b64dd57a917944103bea3a/packages/%40uppy/provider-views/src/ProviderView/AuthView.tsx#L76

Reply:
Will look into what can be done here

Reply:
I think there is no reason for this to be a form at all.
It's a search input, and adorning a `<section/>` tag with `role="search"` is enough for screen readers to understand these are connected elements.

Removed the `<form>` tag, added two `onSubmit` events, one for the button, one for the input: https://github.com/transloadit/uppy/pull/5050/commits/b7e901faaecbda9539b81c1e5301ec3502f0c6a2.

Tested in Firefox, Chrome and Safari;
tested with VoiceOver for accessibility.

@aduh95, @ me if you tried this version before but stumbled upon some issue.

Reply:
I think we do need the `form` attribute, otherwise it's going to mess with the hypothetical user's `<form>` that wraps Uppy.

Reply:
From what I understand, the worry is that the user will submit the form with `<Uppy/>` inside, which would make all input fields automatically submitted, even fields-inside-uppy.
Having a `form={}` attribute supposedly prevents that from happening.

We should check if that's the case, and add a comment if so - otherwise it looks like an accessibility trick that _does not_ improve accessibility upon inspection; or a css trick that can be dealt with in other ways; so developers might be tempted to remove it. Added checking that to my todos.

I'll reintroduce dynamic `<form/>`s in this PR.